src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -121,38 +128,46 @@ public PackageObjectFactory(String packageName, ClassLoader moduleClassLoader) {
      */
     @Override
     public Object createModule(String name) throws CheckstyleException {
-        Object instance = createObjectFromMap(name);
+        final String nameCheck = name + CHECK_SUFFIX;
+        Object instance = createObject(name);
```

Comment:
`createObjectFromMap(name, NAME_TO_FULL_MODULE_NAME);` should be the first thing we try. It will be the fastest and the most common scenario.
pure `createObject(name)` should probably be used last when user supplies full package.

Order of finding module should be, imo: 
1) our module map test
2) 3rd party map test
3) full package (or raw) test
4) fail.

We can probably discuss skipping 1-2 if given name has a period in it.
3 is only really needed if we don't examine maps for the full package. I don't really see a benefit from doing this.